gradle-conjure/src/test/groovy/com/palantir/gradle/conjure/ConjureJavaLocalCodegenPluginIntegrationSpec.groovy

```diff
@@ -139,6 +141,53 @@ class ConjureJavaLocalCodegenPluginIntegrationSpec extends IntegrationSpec {
         sourcesFolderUrls.contains('file://$MODULE_DIR$/src/generated/java')
     }
 
+    def 'embeds product dependencies correctly'() {
+        addSubproject("conjure-api")
+        buildFile << """
+        conjure { java { addFlag 'objects' } }
+        
+        task modifyIr {
+            doFirst {
+                file('build/conjure-ir/conjure-api.conjure.json').text = '''
+                {
+                    "version": "1",
+                    "extensions": {
+                        "recommended-product-dependencies": [{
+                            "product-group": "com.palantir.conjure",
+                            "product-name": "conjure",
+                            "minimum-version": "1.2.0",
+                            "recommended-version": "1.2.0",
+                            "maximum-version": "2.x.x"
+                        }]
+                    }
+                }
+                '''
+            }
+        }
+        
+        modifyIr.mustRunAfter extractConjureIr
+        subprojects {
+            tasks.jar.dependsOn modifyIr
+        }
+        """.stripIndent()
+
+        when:
+        ExecutionResult result = runTasksSuccessfully('jar')
+
+        then:
+        result.wasExecuted(':conjure-api:compileJava')
+        result.wasExecuted(':conjure-api:generateConjure')
+
+        def recommendedProductDependencies = readRecommendedProductDeps(file('conjure-api/build/libs/conjure-api-1.0.0.jar'))
+        recommendedProductDependencies == '{"recommended-product-dependencies":[{' +
```

Comment:
weren't we gonna call this `sls-recommended-product-dependencies` for consistency with what we embed in the jar manifest?

Reply:
The Jar manifest looks like: 
```
Sls-Recommended-Product-Dependencies: {"recommended-product-dependencies
 ":[{"product-group":"com.palantir.food","product-name":"foo-server","mi
 nimum-version":"3.9.0","maximum-version":"3.x.x","recommended-version":
 "3.9.0"}]}
```

So I made it `recommended-predocut-dependencies` consistent with the inner json

Reply:
@iamdanfox We haven't released the change yet so we can make adjustments, but this seemed like a a nice level of consistency